### PR TITLE
0x_api - add bnb - all_txs & deduped & crosschain 

### PR DESCRIPTION
**Pull request by RantumBits**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by RantumBits** [comment]
@jeff-dude this adds a zeroex bnb swaps table - getting these ready for dex/aggregator.trades

**Review by jeff-dude — APPROVED** [review]
pushed a final commit, please confirm if you're good 🤞 
- switch left joins to inner to base tables (some did, some didn't), feel free to revert if left is required on some CTEs
- fixed contributor names to be dune username, please see #3085 where hildobby is looking to fix contributors to receive proper recognition -- 🐐 
- table name fix, assume `fillz` wasn't intentional, but ya never know
- added schema name in project file -- if you trace through the logs in the GH action for CI test, you can find the tables built and see missing schema. you have ~24 hours to validate that test table on dune app

**Comment by jeff-dude** [comment]
> pushed a final commit, please confirm if you're good 🤞
> 
> * switch left joins to inner to base tables (some did, some didn't), feel free to revert if left is required on some CTEs
> * fixed contributor names to be dune username, please see #3085 where hildobby is looking to fix contributors to receive proper recognition -- 🐐
> * table name fix, assume `fillz` wasn't intentional, but ya never know
> * added schema name in project file -- if you trace through the logs in the GH action for CI test, you can find the tables built and see missing schema. you have ~24 hours to validate that test table on dune app

seems like maybe the inner join dropped rows indeed, since seed test now fails? maybe the rows are missing to join to seed?

**Comment by RantumBits** [comment]
this should be good now @jeff-dude  i made some minor updates and updated the seed file, with a successful test in last run. thanks for your help

File: models/zeroex/bnb/zeroex_bnb_api_fills.sql [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+    {% endif %}
+
+),
+uni_v2_swap as (
+SELECT   s.tx_hash tx_hash, s.index evt_index, s.contract_address, s.block_time, 
+    '0x' || substring(DATA, 27, 40) AS maker, 
+    '0xdef1c0ded9bec7f1a1670819833240f027b25eff' AS taker,
+                '0x' || substring(DATA, 91, 40) AS taker_token,
+            '0x' || substring(DATA, 155, 40) AS maker_token,
+            bytea2numeric('0x' || substring(DATA, 219, 40)) AS taker_token_amount_raw,
```
Comment (jeff-dude) [inline models/zeroex/bnb/zeroex_bnb_api_fills.sql:311]: do you want to use v3 function here, like [above](https://github.com/duneanalytics/spellbook/pull/3081/files#diff-da823930ae34f283e8537ff1d497676c1d7bdf32a0069fc643a262222b02ec5bR232)?

File: models/zeroex/bnb/zeroex_bnb_api_fills.sql [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+            bytea2numeric('0x' || substring(DATA, 283, 40)) AS maker_token_amount_raw,
```
Comment (jeff-dude) [inline models/zeroex/bnb/zeroex_bnb_api_fills.sql:312]: same here

**Comment by jeff-dude** [comment]
is there reason you removed from the cross-chain rollup spell? do you prefer to merge and iterate prior to adding?

File: models/zeroex/bnb/zeroex_bnb_api_fills.sql (hunk 1 above)
Comment (RantumBits) [reply models/zeroex/bnb/zeroex_bnb_api_fills.sql]: nice catch

File: models/zeroex/bnb/zeroex_bnb_api_fills.sql (hunk 2 above)
Comment (RantumBits) [reply models/zeroex/bnb/zeroex_bnb_api_fills.sql]: updated both instances 👌

**Comment by RantumBits** [comment]
> is there reason you removed from the cross-chain rollup spell? do you prefer to merge and iterate prior to adding?

added back in and getting an error on DuneSQL check - same as on #3110 . What's best way to handle?

**Comment by jeff-dude** [comment]
> > is there reason you removed from the cross-chain rollup spell? do you prefer to merge and iterate prior to adding?
> 
> added back in and getting an error on DuneSQL check - same as on #3110 . What's best way to handle?

since the cross-chain rollup spell is a view, dunesql will be the engine to run against it. `array()` function is not on trino, only spark. i'm moving all transformations to the incremental models, which still runs on spark (for the time being). it's also best practice for us to move away from using in the cross-chain views, but rather at the base tables. this should fix it

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**